Add bring your own snapshot documentation to configuration-options.md (#9573)
opensearch-project/documentation-website: _migration-assistant/deploying-migration-assistant/configuration-options.md

All of these authentication options apply to both source and target clusters.

## Bring your own snapshot options

> An existing snapshot can alternatively be used to perform metadata as well as backfill migrations, instead of using the Migration Assistant to create a snapshot.

Reviewer: ```suggestion
An existing Amazon S3 snapshot can alternatively be used to perform metadata as well as backfill migrations, instead of using the Migration Assistant to create a snapshot.
```

All of these authentication options apply to both source and target clusters.

## Bring your own snapshot options

> An existing snapshot can alternatively be used to perform metadata as well as backfill migrations, instead of using the Migration Assistant to create a snapshot.

Reviewer 2: At this point, do we want to specify non-incremental snapshots, or are we okay with either kind?

Author: After discussing, we should be able to handle both but want to add further testing for incremental snapshots in our repo

```json
    "snapshot": {
        "snapshotName": "my-snapshot-name",
        "s3Uri": "s3://my-s3-bucket-name/my-bucket-path-to-snapshot-repo",
        "s3Region": "us-east-2"
    }
```
{% include copy.html %}

> As a default, S3 buckets will automatically allow roles in the same account (with the appropriate `s3:*` permissions) to access the S3 bucket regardless of region. So if the external S3 bucket being utilized is in the same account as the Migration Assistant deployment, no IAM configuration may be required to access this bucket.

Reviewer 2: nit: `no custom IAM configuration` or `no further IAM configuration`

> As a default, S3 buckets will automatically allow roles in the same account (with the appropriate `s3:*` permissions) to access the S3 bucket regardless of region. So if the external S3 bucket being utilized is in the same account as the Migration Assistant deployment, no IAM configuration may be required to access this bucket.

Reviewer: ```suggestion
As a default, S3 buckets will automatically allow roles in the same account (with the appropriate `s3:*` permissions) to access the S3 bucket regardless of region. If the external S3 bucket being utilized is in the same account as the Migration Assistant deployment, no further IAM configuration is required to access this bucket.
```

All of these authentication options apply to both source and target clusters.

## Bring your own snapshot options

> An existing Amazon S3 snapshot can alternatively be used to perform metadata as well as backfill migrations, instead of using the Migration Assistant to create a snapshot.

Reviewer: They are two different flavors of migrations. I added links to the pages related to both.
